language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -528,92 +570,74 @@ writePackageIdTs dir pkgId =
       ["export default '" <> unPackageId pkgId <> "';"]
 
 writeTsConfig :: FilePath -> IO ()
-writeTsConfig dir = writeFileUTF8 (dir </> "tsconfig.json") $ unlines
-    [ "{"
-    , "  \"compilerOptions\": {"
-    , "    \"target\": \"es5\","
-    , "    \"lib\": ["
-    , "      \"dom\","
-    , "      \"es2015\""
-    , "     ],"
-    , "    \"strict\": true,"
-    , "    \"noUnusedLocals\": true,"
-    , "    \"noUnusedParameters\": false,"
-    , "    \"noImplicitReturns\": true,"
-    , "    \"noFallthroughCasesInSwitch\": true,"
-    , "    \"outDir\": \"lib\","
-    , "    \"module\": \"commonjs\","
-    , "    \"declaration\": true,"
-    , "    \"sourceMap\": true"
-    , "  },"
-    , "  \"include\": [\"src/**/*.ts\"],"
-    , "}"
-    ]
+writeTsConfig dir =
+  BSL.writeFile (dir </> "tsconfig.json") $ encodePretty tsConfig
+  where
+    tsConfig :: Value
+    tsConfig = object
+      [ "compilerOptions" .= object
+        [ "target" .= ("es5" :: T.Text)
```

Comment:
Do we really need to mark this a `T.Text`? Isn't `"es5"` just fine? That will give us a `String` and the right thing.
```suggestion
        [ "target" .= "es5"
```
Same in many places below.

Reply:
It really needs to be marked as Text.

Reply:
Ah right, because we have `OverloadedStrings` turned on to be able to get `Text` literals. 😦